Uber mock, update golangci-lint

lovromazgon opened the pull request:
### Description

- Replace `github.com/golang/mock` with `go.uber.org/mock` (it was deprecated)
- Add depguard rule to prevent further use of `github.com/golang/mock`
- Update golangci-lint to 1.53.3

maha-hajja reviewed (approved):
LGTM! maybe it's a good idea to add the workflow [validate-generated-files](https://github.com/ConduitIO/conduit/blob/main/.github/workflows/validate-generated-files.yml) to the repo too

lovromazgon commented:
@maha-hajja good point, I added the CI action and some other things I noticed:
- dependabot config
- updated dependencies to latest versions
- added buf as a dependency in `tools.go`
- added a makefile

This should clean up most things in this repo.

maha-hajja reviewed (approved):
perfect! looks nice
